diff --git a/dlls/win32u/vulkan.c b/dlls/win32u/vulkan.c
--- a/dlls/win32u/vulkan.c
+++ b/dlls/win32u/vulkan.c
@@ -3004,7 +3004,14 @@ static void vulkan_init_once(void)
     if (!(libvulkan = getenv( "CX_LIBVULKAN" )) ||
         !(graphics_backend = getenv( "CX_ACTIVE_GRAPHICS_BACKEND" )) ||
         strcmp( graphics_backend, "wined3d" ))
+#ifdef SONAME_LIBVULKAN
         libvulkan = SONAME_LIBVULKAN;
+#else
+    {
+        ERR( "Vulkan support was not built into this runtime\n" );
+        return;
+    }
+#endif
     else
         ERR( "Using %s\n", libvulkan );
 
