diff --git a/dlls/ntdll/unix/loader.c b/dlls/ntdll/unix/loader.c
--- a/dlls/ntdll/unix/loader.c
+++ b/dlls/ntdll/unix/loader.c
@@ -2242,21 +2242,6 @@ static void start_main_thread(void)
     hook(localtime, my_localtime);
 #endif
 
-    /* CW Hack 24067 */
-    {
-        void *cxcompatdb = NULL;
-        char *name = NULL;
-
-        asprintf( &name, "%s/cxcompatdb.so", ntdll_dir );
-        if (name)
-        {
-            cxcompatdb = dlopen( name, RTLD_LOCAL | RTLD_LAZY );
-            if (!cxcompatdb)
-                WARN( "error loading cxcompatdb.so: %s\n", dlerror() );
-            free(name);
-        }
-    }
-
     server_init_process_done();
 }
 
